mocks/consumer_service.go

```diff
@@ -0,0 +1,42 @@
+package mocks
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/medibloc/panacea-oracle/consumer_service"
+)
+
+type MockConsumerService struct {
+	oraclePrivKey *btcec.PrivateKey
+}
+
+var (
+	_           consumer_service.FileStorage = &MockConsumerService{}
+	defaultPath                              = "MockConsumerService"
+)
+
+func (u MockConsumerService) Add(endpoint string, dealID uint64, dataHash string, data []byte) error {
+	if err := os.MkdirAll(filepath.Join(defaultPath, strconv.FormatUint(dealID, 10)), fs.ModePerm); err != nil {
+		return err
+	}
+
+	if err := os.WriteFile(filepath.Join(defaultPath, strconv.FormatUint(dealID, 10), dataHash), data, fs.ModePerm); err != nil {
+		return err
+	}
```

Comment:
I don't have any preference of either storing file to the filesystem or just holding it in memory. It's up to you depending on test scenarios.
If you want to store file in disk, you can use tmp directories (`/tmp` in linux and `/var/folders/...` in mac) that is cleaned automatically by OS if you don't clean it manually. 

Go's `testing` package provides a magical function [testing.T.TempDir()](https://pkg.go.dev/testing#T.TempDir) that creates a tmp dir with a random unique name and removes it when the test is finalized. But, if you cannot access `testing.T` in this mock file, you can use [os.MkdirTemp()](https://pkg.go.dev/os#MkdirTemp).